### ISSUE TITLE
Mix-in enum members have parent type when collected

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1351]
**Describe the bug**
Members of an `Enum` declared with a mix-in type inherit the mix-in type when the `Enum` is cast to a built-in collection.  They should have the `Enum` type.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of `list(Foo)` would be `list[Foo]` - not `list[str]`.

**Screenshots or Code**
```python
from enum import Enum

class Foo(str, Enum):
    bar = 'bar'

for member in Foo:
    reveal_type(member)  # Ok: 'Type of "member" is "Foo"'

foo_members = list(Foo)
reveal_type(foo_members)  # Type of "foo_members" is "list[str]"
print([type(i) for i in foo_members])  # [<enum 'Foo'>]

```

**VS Code extension or command-line**
Either, 1.1.99

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
